### ISSUE TITLE
Incorrect shader processing in GLTFLoader v2 

### ISSUE THREAD

**Issue by proteamer** [mrdoob/three.js#9890]
##### Description of the problem 

There is the part of shader which was generated via [gltf-pipeline](https://github.com/AnalyticalGraphicsInc/gltf-pipeline) with `--encodeNormals` flag:

```
attribute vec2 a_normal;
vec3 gltf_oct_dec_a_normal;

uniform mat4 u_modelViewMatrix;
uniform mat4 u_projectionMatrix;
uniform mat3 u_normalMatrix;
attribute vec3 a_position;
varying vec3 v_positionEC;

varying vec3 v_normal;

void gltf_oct_dec_normal()
{
  vec4 pos = u_modelViewMatrix * vec4(a_position,1.0);
  v_positionEC = pos.xyz;
  gl_Position = u_projectionMatrix * pos;
  v_normal = u_normalMatrix * gltf_oct_dec_a_normal;
}

void main() {
    gltf_oct_dec_a_normal = gltf_octDecode_vec2(a_normal);
    gltf_oct_dec_normal();
}
```

The issue is that [replaceTHREEShaderAttributes](https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/GLTFLoader.js#L626) replaces all `a_normal` substrings with `normal` in shader code. So it affects all varriables which names include `a_normal` for instance `vec3 gltf_oct_dec_a_normal;` will replaced with `vec3 gltf_oct_dec_normal;` and in cause `redefinition function` error because threre is `void gltf_oct_dec_normal()` function with the same name.

##### Three.js version

- [x] Dev
- [ ] r81
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

##### Hardware Requirements (graphics card, VR Device, ...)

It doesn't matter for this issue.

**Comment by proteamer** [comment]
I've prepared my fix in #9895 please check it.

**Comment by richtr** [comment]
Oops! I didn't see your update to this issue @proteamer and I also created a PR to fix this :/ 

https://github.com/mrdoob/three.js/pull/9897
